lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,263 @@
+        will be. Given that, it is best to use the http(s) url for I(state=present)
+        and the reverse DNS I(state=absent). Alternatively reverse dns format can optimally
+        be used with I(state=absent), ex. I(name=org.gnome.gedit).
+    required: true
+  remote:
+    description:
+      - The flatpak I(remote) repo to be used in the flatpak operation.
+    default: 'None'
+  state:
+    description:
+      - Set to C(present) will install the flatpak and/or I(remote).
+        Set to C(absent) will remove the flatpak and/or I(remote).
+    default: present
+    choices: [ absent, present ]
+'''
+EXAMPLES = '''
+ - name: Install the spotify flatpak
+   flatpak:
+    name:  https://s3.amazonaws.com/alexlarsson/spotify-repo/spotify.flatpakref
+    state: present
+
+ - name: Add the gnome remote andd install gedit flatpak
+   flatpak:
+    name: https://git.gnome.org/browse/gnome-apps-nightly/plain/gedit.flatpakref
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    state: absent
+
+ - name: Remove the gedit flatpak and remote
+   flatpak:
+    name: org.gnome.gedit
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    state: absent
+
+ - name: Remove the gedit package
+   flatpak:
+    name: org.gnome.gedit
+    state: absent
+'''
+
+RETURN = '''
+reason:
+    description: On failure, the output for the failure
+    returned: failed
+    type: string
+    sample: error while installing...
+name:
+    description: Name of flatpak given for the operation
+    returned: always
+    type: string
+    sample: https://git.gnome.org/.../gnome-apps/gedit.flatpakref
+remote:
+    description: Remote of flatpak given for the operation
+    returned: always
+    type: string
+    sample: https://sdk.gnome.org/gnome-apps.flatpakrepo
+'''
+
+
+import subprocess
+from ansible.module_utils.basic import AnsibleModule
+
+
+def install_flat(binary, flat, module):
+    if module.check_mode:
+        # Check if any changes would be made but don't actually make
+        # those changes
+        module.exit_json(changed=True)
+
+    command = "{} install -y --from {}".format(binary, flat)
+
+    output = flatpak_command(command)
+    if 'error' in output and 'already installed' not in output:
+        return 1, output
+
+    return 0, output
+
+
+def uninstall_flat(binary, flat, module):
+    # This is a difficult function because it seems there
+    # is no naming convention for the flatpakref to what
+    # the installed flatpak will be named.
+    common_name = parse_flat(flat)
+    if module.check_mode:
+        # Check if any changes would be made but don't actually make
+        # those changes
+        module.exit_json(changed=True)
+
+    command = "{} list --app".format(binary)
+    process = subprocess.Popen(
+        command.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = subprocess.check_output(
+        ("grep", "-i", common_name), stdin=process.stdout)
```

Comment:
Why run grep and not parse it yourself in python ?

Reply:
valid point.